src/share/poudriere/common.sh

```diff
@@ -2940,6 +2940,29 @@ jail_cleanup() {
 	export CLEANED_UP=1
 }
 
+download_from_repo() {
+	msg "Prefetching missing packages from pkg+http://pkg.freebsd.org/\${ABI}/${PACKAGE_BRANCH}"
+	cat >> ${MASTERMNT}/etc/pkg/poudriere.conf <<EOF
+FreeBSD: {
+        url: pkg+http://pkg.freebsd.org/\${ABI}/${PACKAGE_BRANCH};
+}
+EOF
+	umount ${UMOUNT_NONBUSY} ${MASTERMNT}/packages || \
+	    umount -f ${MASTERMNT}/packages
+	mount_packages
+	# only list packages which do not exists to prevent pkg from overwriting prebuilt packages
+	# XXX only work when PKG_EXT is the same as the upstream
+	(
+	while mapfile_read_loop "all_pkgs" pkgname originspec _ignored; do
+		[ -f ${MASTERMNT}/packages/All/${pkgname}.${PKG_EXT} ] || echo ${pkgname}
+	done
+	)| JNETNAME="n" injail xargs env -i ASSUME_ALWAYS_YES=yes pkg fetch -o /packages
```

Comment:
I think this is fine for now but ultimately we may want to inspect more what will be downloaded to avoid `delete_old_pkg` from nuking the same packages every time for things like different options or flavors.